ui-components/src/forms/Field.tsx

```diff
@@ -57,20 +57,42 @@ const Field = (props: FieldProps) => {
   if (props.bordered && (props.type === "radio" || props.type === "checkbox"))
     controlClasses.push("field-border")
 
-  const formatValue = () => {
+  const formatValue = (focused = false) => {
     if (props.getValues && props.setValue) {
       const currencyValue = props.getValues(props.name)
       const numericIncome = parseFloat(currencyValue)
-      if (!isNaN(numericIncome)) {
-        props.setValue(props.name, numericIncome.toFixed(2))
+
+      if (focused && currencyValue) {
+        props.setValue(props.name, parseFloat(currencyValue.replaceAll(",", "")))
+      } else if (isNaN(numericIncome)) {
+        props.setValue(props.name, "")
+      } else {
+        props.setValue(
+          props.name,
+          numericIncome.toLocaleString("en-US", { minimumFractionDigits: 2 })
+        )
       }
     }
   }
 
+  const filterNumbers = (e: ChangeEvent<HTMLInputElement>) => {
+    if (props.setValue) {
+      props.setValue(props.name, e.target.value.replace(/[a-z]|[A-Z]/g, "").match(/^\d*\.?\d?\d?/g))
+    }
+  }
+
   let inputProps = { ...props.inputProps }
-  if (props.type === "currency") inputProps = { ...inputProps, step: 0.01, onBlur: formatValue }
+  if (props.type === "currency") {
+    inputProps = {
+      ...inputProps,
+      step: 0.01,
```

Comment:
I'm not noticing a step functionality in the field. I know you didn't add this, but curious if it is needed anymore?